curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md

```diff
@@ -17,439 +17,445 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --distractors--
 
-Placeholder distractor 1
+To modify keyframe sequences based on variable values
 
 ---
 
-Placeholder distractor 2
+To control custom property inheritance during animation
 
 ---
 
-Placeholder distractor 3
+To ensure that custom properties maintain compatibility with media queries
 
 #### --answer--
 
-Placeholder answer
+To define which custom properties are allowed to animate and set their initial values
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Why is it important to use fallback values when applying custom properties in CSS?
 
 #### --distractors--
 
-Placeholder distractor 1
+To avoid undefined values in unsupported browsers
 
 ---
 
-Placeholder distractor 2
+To improve performance
 
 ---
 
-Placeholder distractor 3
+To optimize rendering time
 
 #### --answer--
 
-Placeholder answer
+To reduce code complexity
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In which scenario might defining a custom property in a media query cause unexpected behavior?
 
 #### --distractors--
 
-Placeholder distractor 1
+When the property is defined using a `px` unit
 
 ---
 
-Placeholder distractor 2
+When the property is defined outside of the `:root`
 
 ---
 
-Placeholder distractor 3
+When the property is defined at the bottom of the stylesheet
 
 #### --answer--
 
-Placeholder answer
+When the media query condition no longer applies, and the custom property is no longer valid
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What’s a key drawback when using CSS variables in combination with older layout models like `float` or `table`?
 
 #### --distractors--
 
-Placeholder distractor 1
+They cannot be referenced by the `float` property
 
 ---
 
-Placeholder distractor 2
+They may cause significant reflows, impacting performance
 
 ---
 
-Placeholder distractor 3
+They require JavaScript to function correctly
 
 #### --answer--
 
-Placeholder answer
+Older layout models may not properly inherit or apply custom properties, causing inconsistencies
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When using custom properties with CSS Grid or Flexbox, what should you be cautious about?
 
 #### --distractors--
 
-Placeholder distractor 1
+Custom properties override grid or flex properties by default
 
 ---
 
-Placeholder distractor 2
+Flexbox and Grid ignore custom properties during layout calculations
 
 ---
 
-Placeholder distractor 3
+Custom properties can cause layout shifts when recalculated
 
 #### --answer--
 
-Placeholder answer
+Custom properties should be managed carefully, as they are recalculated with each layout pass, potentially impacting performance
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+If you define a custom property `--text-color` inside a media query and use `var(--text-color)` in multiple elements, what happens if the viewport changes and the media query no longer matches?
```

Comment:
There's previous very similar quesion:
> When defining a variable `--foreground` inside a media query, what happens when the media query no longer matches the current viewport?
> - The custom property is preserved and continues to apply
> - The custom property reverts to its initial value
> - The custom property is recalculated based on the viewport
> - The custom property is no longer available as it is scoped to the media query

Reply:
resolved in new commit (changed question)